### DISCUSSION TITLE
VRPTW: Minimizing total driving time

### DISCUSSION THREAD

**Discussion by amih77** [google/or-tools#2969] [Routing (and legacy CP) questions]
The objective of the problem I'm trying to solve is to minimize the total sum of driving durations of the vehicles under time windows constraints. I'm struggling to find the right setup in order to do so. 

I have 2 problematic scenarios which I can't solve at once:

1. Consider 3 vehicles and 5 tasks and duration matrix (task locations first, then start locations and end locations):
```
[
	[0, 1556, 1959, 2672, 3214, 17171, 17171, 17171, 2253, 2253, 2253], 
	[17171, 0, 1003, 1818, 2360, 17171, 17171, 17171, 1876, 1876, 1876], 
	[17171, 17171, 0, 1415, 2146, 17171, 17171, 17171, 3044, 3044, 3044], 
	[17171, 17171, 17171, 0, 1331, 17171, 17171, 17171, 3448, 3448, 3448], 
	[17171, 17171, 17171, 17171, 0, 17171, 17171, 17171, 1946, 1946, 1946], 
	[810, 1465, 1486, 2199, 2741, 17171, 17171, 17171, 600, 600, 600], 
	[810, 1465, 1486, 2199, 2741, 17171, 17171, 17171, 600, 600, 600], 
	[810, 1465, 1486, 2199, 2741, 17171, 17171, 17171, 600, 600, 600], 
	[17171, 17171, 17171, 17171, 17171, 17171, 17171, 17171, 17171, 17171, 17171], 
	[17171, 17171, 17171, 17171, 17171, 17171, 17171, 17171, 17171, 17171, 17171], 	
	[17171, 17171, 17171, 17171, 17171, 17171, 17171, 17171, 17171, 17171, 17171]
]

```
(here, 17171 is just a dummy value for pairs that are not interesting in terms of time window (`a -> b` where `a` needs to be visited after `b`), `location -> start location` or `end location - > any other location.`)

The time windows are (locations first, then start locations and end locations):
`[[3600, 4020], [8400, 8820], [14400, 14820], [21600, 22020], [25800, 26220], [119, 43319], [119, 43319], [10919, 54119], [121, 43320], [121, 43320], [10921, 54120]]`

I'm expecting vehicle 0 or vehicle 1 to be assigned with all tasks as it is a feasible solution and it's minimal. Indeed , when I set the strategy to `PARALLEL_CHEAPEST_INSERTION` I get:
```
5 Time(119,3210) -> 0 Time(3600,4020) -> 1 Time(8400,8820) -> 2 Time(14400,14820) -> 3 Time(21600,22020) -> 4 Time(25800,26220) -> 8 Time(27746,43320)
Time of the route: 27746sec
```
However, when I try with `PATH_CHEAPEST_ARC` I get:
```
Route for vehicle 0:
5 Time(119,3210) -> 0 Time(3600,4020) -> 1 Time(8400,8820) -> 2 Time(14400,14820) -> 8 Time(17444,43320)
Time of the route: 17444sec

Route for vehicle 1:
6 Time(119,19821) -> 3 Time(21600,22020) -> 9 Time(25048,43320)
Time of the route: 25048sec

Route for vehicle 2:
7 Time(10919,23479) -> 4 Time(25800,26220) -> 10 Time(27746,54120)
Time of the route: 27746sec

Total time of all routes: 70238sec.

```
So I set the strategy to `PARALLEL_CHEAPEST_INSERTION`.

Now, consider problem 2: 3 vehicles, 2 tasks, durations matrix:
```
[
	[0, 4082, 17171, 17171, 17171, 3997, 3300, 3997], 
	[17171, 0, 17171, 17171, 17171, 3997, 3300, 3997], 
	[2610, 2610, 17171, 17171, 17171, 2400, 3265, 2400], 
	[2610, 2610, 17171, 17171, 17171, 2400, 3265, 2400], 
	[2400, 2400, 17171, 17171, 17171, 2559, 3356, 2559], 
	[17171, 17171, 17171, 17171, 17171, 17171, 17171, 17171], 
	[17171, 17171, 17171, 17171, 17171, 17171, 17171, 17171], 
	[17171, 17171, 17171, 17171, 17171, 17171, 17171, 17171]
]
```

and time windows:
`[[4500, 4920], [12900, 13320], [119, 32519], [119, 8519], [9419, 32519], [121, 32520], [121, 8520], [9421, 32520]]
`
I'm expecting task 0 to vehicle 1 and task 1 to vehicle 2, and it's what I get under `PATH_CHEAPEST_ARC`:
```
Route for vehicle 0:
2 Time(119,30120) -> 5 Time(2519,32520)
Time of the route: 2519sec

Route for vehicle 1:
3 Time(119,2310) -> 0 Time(4500,4920) -> 6 Time(7800,8520)
Time of the route: 7800sec

Route for vehicle 2:
4 Time(9419,10920) -> 1 Time(12900,13320) -> 7 Time(16897,32520)
Time of the route: 16897sec

```
However, under `PARALLEL_CHEAPEST_INSERTION` (which solves scenario 1) I get:
```
Route for vehicle 0:
2 Time(119,2310) -> 0 Time(4500,4920) -> 1 Time(12900,13320) -> 5 Time(16897,32520)
Time of the route: 16897sec

Route for vehicle 1:
3 Time(119,5255) -> 6 Time(3384,8520)
Time of the route: 3384sec

Route for vehicle 2:
4 Time(9419,29961) -> 7 Time(11978,32520)
Time of the route: 11978sec
```

which is not minimal in the sense I'm looking for.

I have tried experimenting with `SetFixedCostOfAllVehicles(10000)` but wasn't successful with it. 

To summarize - I am looking for the right parameters and initialization that will minimize the driving time of the vehicles, anything else doesn't matter for this problem. 

Thanks in advance for any help!
